app/carbonzipper/http_handlers.go

```diff
@@ -34,12 +35,65 @@ const (
 	formatTypeProtobuf3 = "protobuf3"
 )
 
+func findErrorsFanIn(ctx context.Context, errs []error, numBackends int, logger *zap.Logger) error {
+	nErrs := len(errs)
+	switch {
+	case (nErrs == 0):
+		return nil
+	case (nErrs < numBackends):
+		// if at least one backend is successful, it's a success! But log a warning
+		logger.Warn("Some requests failed with generic errors",
+			zap.String("uuid", util.GetUUID(ctx)),
+			zap.String("summary", errorsSummary(errs)),
+		)
+		return nil
+	default:
+		// everything failed.
+		// If all the failures are not-founds, it's a not-found
+		allErrorsNotFound := true
+		for _, e := range errs {
+			if _, ok := e.(types.ErrNotFound); !ok {
+				allErrorsNotFound = false
+				break
+			}
+		}
+
+		if allErrorsNotFound {
+			return types.ErrNotFound("all backends returned not found")
+		}
+
+		// if it's not a not-found, it's an internal error
+		logger.Error("Requests from all backends failed",
```

Comment:
Shouldn't it log the this first, before the return above.

Reply:
No, because we only want the log when we return generic error. This way we have no error log when it's a not found, which is good. This way we get much cleaner logs.